packages/cactus-core/src/test/typescript/unit/plugin-registry.test.ts

```diff
@@ -46,63 +46,30 @@ test("PluginRegistry", (tMain: Test) => {
     ],
   });
 
-  test("findOneByKeychainId() finds plugin by keychain ID", (t: Test) => {
-    t.doesNotThrow(() => pluginRegistry.findOneByKeychainId(keychainId));
+  test("findOneByKeychainId() finds plugin by keychain ID", () => {
+    expect(() => pluginRegistry.findOneByKeychainId(keychainId)).not.toThrow();
     const keychainPlugin = pluginRegistry.findOneByKeychainId(keychainId);
-    t.equal(keychainPlugin, mockKeychainPlugin, "Finds same object OK");
+    expect(keychainPlugin).toEqual(mockKeychainPlugin);
 
-    t.throws(
-      () => pluginRegistry.findOneByKeychainId(""),
-      /need keychainId arg as non-blank string/,
-      "Check for keychain ID blankness OK",
-    );
-    t.throws(
-      () => pluginRegistry.findOneByKeychainId("x"),
-      /No keychain found for ID/,
-      "Throws for keychain not found OK",
-    );
-
-    t.end();
+    expect(() => pluginRegistry.findOneByKeychainId("")).toThrow();
```

Comment:
@awadhana Please update the throw checks to also verify the message that's being thrown, see examples here: https://stackoverflow.com/questions/46042613/how-to-test-the-type-of-a-thrown-exception-in-jest/46155381#46155381

Reply:
@petermetz  this is the OG pr with the .toThrow() and .not.toThrow() (in place of doesNotThrow) that I'm having an issue keeping the message with. I tried the Jest file that you've mentioned as well as https://jestjs.io/docs/expect#tothrowerror which has a few different ways to throw an error depending on how strict the equal is for the error message. I've emailed you what I've done and the error message I'm receiving. The email header is "Q and A Thread".

Reply:
@awadhana Thank you, I'll respond to the email. For next time it's better to post that info right here so that we can refer to it later via public URLs (which cannot be done with internal email communication unfortunately).

Reply:
@awadhana Sent response to the email, pasting the working code here as well:

```typescript
import { v4 as uuidv4 } from "uuid";
import "jest-extended";

import { PluginRegistry } from "../../../main/typescript/public-api";

import { ICactusPlugin, IPluginKeychain } from "@hyperledger/cactus-core-api";

describe("PluginRegistry", () => {
  const keychainId = uuidv4();
  const instanceId = uuidv4();

  const mockKeychainPlugin: IPluginKeychain = {
    getInstanceId: () => instanceId,
    getKeychainId: () => keychainId,
    delete: async () => {
      throw new Error("This is a mock. Not implemented.");
    },
    has: async () => {
      throw new Error("This is a mock. Not implemented.");
    },
    get: async () => {
      throw new Error("This is a mock. Not implemented.");
    },
    set: async () => {
      throw new Error("This is a mock. Not implemented.");
    },
    getPackageName: () => "@hyperledger/cactus-plugin-keychain-mock",

    onPluginInit: async () => {
      throw new Error("not sure how this works");
    },
  };

  const pluginRegistry = new PluginRegistry({
    plugins: [
      mockKeychainPlugin,
      {
        getInstanceId: () => "some-mock-plugin-instance-id-1",
      } as ICactusPlugin,
      {
        getInstanceId: () => "some-mock-plugin-instance-id-2",
      } as ICactusPlugin,
      {
        getInstanceId: () => "some-mock-plugin-instance-id-3",
      } as ICactusPlugin,
    ],
  });

  test("findOneByKeychainId() finds plugin by keychain ID", () => {
    expect(() => pluginRegistry.findOneByKeychainId(keychainId)).not.toThrow();
    const keychainPlugin = pluginRegistry.findOneByKeychainId(keychainId);
    expect(keychainPlugin).toEqual(mockKeychainPlugin);
    expect(() => pluginRegistry.findOneByKeychainId("")).toThrowError(
      new Error(
        `PluginRegistry#findOneByKeychainId() need keychainId arg as non-blank string.`,
      ),
    );
    expect(() => pluginRegistry.findOneByKeychainId("x")).toThrowError(
      new Error(
        `PluginRegistry#findOneByKeychainId() No keychain found for ID x`,
      ),
    );
  });

  test("findOneById() finds plugin by its instanceID", () => {
    expect(() => pluginRegistry.findOneById(instanceId)).not.toThrow();
    const keychainPlugin = pluginRegistry.findOneById(instanceId).get();
    expect(keychainPlugin).toEqual(mockKeychainPlugin);
    expect(() => pluginRegistry.findOneById("")).toThrowError(
      new Error(`"instanceId" is a blank string. Need non-blank.`),
    );

    expect(pluginRegistry.findOneById("x").isEmpty()).toBe(true);
  });

  test("getOneById() finds plugin by its instanceID", () => {
    expect(() => pluginRegistry.getOneById(instanceId)).not.toThrow();
    const keychainPlugin = pluginRegistry.getOneById(instanceId);
    expect(keychainPlugin).toEqual(mockKeychainPlugin);
    expect(() => pluginRegistry.getOneById("")).toThrowError(
      new Error(`"instanceId" is a blank string. Need non-blank.`),
    );
    expect(() => pluginRegistry.getOneById("x")).toThrowError(
      new Error("Plugin x not present in registry"),
    );
  });
});

```